### PR TITLE
job-manager: send events stream immediately

### PR DESCRIPTION
**Pull request by chu11**
Built on top of #3236 .

Per #3245 (but mostly discussed in #3209), this refactors the job-manager to send out `job-manager.events` immediately, not waiting for the KVS commit of eventlog entries to complete.

### CODE REVIEW

File: src/modules/job-manager/event.c [hunk 1]
```diff
@@ -81,6 +80,42 @@ struct events_listener {
...
+    if (flux_msg_pack (msg, "{s:O}", "events", el->events) < 0) {
+        flux_log_error (ctx->h, "%s: flux_msg_pack",
+                        __FUNCTION__);
+        goto out;
+    }
+    if (flux_send (ctx->h, msg, 0) < 0) {
+        flux_log_error (ctx->h, "%s: error sending batch response",
+                        __FUNCTION__);
+        goto out;
+    }
```
Comment (grondo) [inline src/modules/job-manager/event.c:101]: Could `flux_respond_pack(3)` be used here?

File: src/modules/job-manager/event.c [hunk 2]
```diff
@@ -125,6 +160,21 @@ static void event_batch_commit (struct event *event)
 
     if (batch) {
         event->batch = NULL;
+        /* send listener responses now, do not wait on the KVS commit
+         * and before any additional events could be added onto
+         * listeners via event_batch_process_event_entry()
```
Comment (grondo) [inline src/modules/job-manager/event.c:165]: Just noticed this comment reads like a word or line is missing.

**Review by grondo — COMMENTED** [review]
I don't have enough familiarity with job-info or job-manager to make any substantive comments, but I noticed a couple very minor things while perusing the diff.

**Comment by chu11** [comment]
@grondo thanks, re-pushed, fixed up those few things you mentioned. went ahead and squashed since its pretty tiny.

**Comment by chu11** [comment]
given discussion in #3261 closing

**Pull request closed without merging**